tfrecorder/types.py

```diff
@@ -15,13 +15,74 @@
 # limitations under the License.
 
 """Custom types."""
+import dataclasses
+from typing import Tuple, List, Any
 
-from typing import Tuple
-
+import tensorflow as tf
 from apache_beam.pvalue import PCollection
 from tensorflow_transform import beam as tft_beam
 
-
 BeamDatasetMetadata = tft_beam.tft_beam_io.beam_metadata_io.BeamDatasetMetadata
 TransformedMetadata = BeamDatasetMetadata
 TransformFn = Tuple[PCollection, TransformedMetadata]
+
+@dataclasses.dataclass
+class SupportedType:
+  """Base type for TFRecorder Types."""
+  feature_spec: tf.io.FixedLenFeature
+  allowed_values: List[Any]
+
+@dataclasses.dataclass
+class ImageUri(SupportedType):
+  """Supports image uri columns."""
+  feature_spec=tf.io.FixedLenFeature([], tf.string)
+  allowed_values=[]
+
+@dataclasses.dataclass
+class SplitKey(SupportedType):
+  """Supports split key columns."""
+  feature_spec=tf.io.FixedLenFeature([], tf.string)
+  allowed_values = ['TRAIN', 'VALIDATION', 'TEST', 'DISCARD']
+
+@dataclasses.dataclass
+class IntegerInput(SupportedType):
+  """Supports integer columns."""
+  feature_spec=tf.io.FixedLenFeature([], tf.int64)
+  allowed_values=[]
+
+@dataclasses.dataclass
+class FloatInput (SupportedType):
+  """Supports float columns."""
+  feature_spec=tf.io.FixedLenFeature([], tf.float64)
```

Comment:
Change to `tf.float32`. TFT doesn't support `tf.float64` and will return an error: https://github.com/tensorflow/transform/blob/master/tensorflow_transform/tf_metadata/schema_utils.py#L339

Reply:
Fixed.  I'll test 64 later.